### ISSUE TITLE
Router.refresh() called in nested route segment does not refresh the parent route segment.

### ISSUE THREAD

**Issue by dream-ellie** [vercel/next.js#46949]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Sat Jun 18 17:05:47 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.14.0
      npm: 8.5.0
      Yarn: 1.22.19
      pnpm: 6.11.0
    Relevant packages:
      next: 13.2.4-canary.8
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-hid1qx?file=app/page.tsx

### To Reproduce

1. Create a NextJs application that uses the `app` directory
2. Define a folder structure that would have a nested routes. E.g:
```
/app
    page.tsx
    /pages/
        [slug]
           page.tsx
```
4. Make sure all pages are fetching the data dynamically to enable dynamic rendering (use `fetch('...', { cache: 'no-store' })`)
3. Add a button to the nested route page `/app/pages/[slug]/page.tsx` that modifies/mutates the data that impacts both current route and the parent route and calls `router.refresh()`
> _For example, parent page may show list of invoices total amount per invoice and current route shows detailed view of an invoice and mutation changes the total amount of the invoice; so the amount should also change in the invoice list view)._
5. Run the app and click the button to see the result of data update reflected in both nested route and its parent.

### Describe the Bug

**TL;DR**

When calling `router.refresh()` after data mutation (like recommended in the docs: https://beta.nextjs.org/docs/data-fetching/mutating) t**he current route gets updated with the fresh data but the parent route segment is not**.

**Details**
When we have a parent route segment and a child route segment and the data mutation (and `router.refresh()`) happens in the child segment, the parent segment does not get updated data from the server.

A practical example of this could be an invoice list that shows list of invoices with the amounts for each invoice: `/invoices`.
If we have a nested route for showing detailed view that allow editing the amount like so: `/invoices/[invoiceId]`
Then the change in the invoice amount is not reflected in the list view after the update.

The documentation on this page (https://beta.nextjs.org/docs/data-fetching/mutating) says the following:
>By calling router.refresh(), the current route will refresh and fetch an updated list of todos from the server. This does not affect browser history, but **it does refresh data from the root layout down**. 

This sentence leads to thinking that the entire route tree will get fresh data from the server but obviously this is not the case.

### Expected Behavior

There should be one of the two outcomes to resolve this issue:

**Option 1 (current behaviour is a bug):**
Then it should be fixed and the parent route segment should be updated with the fresh data.

**Option 2 (this is working as expected/designed):**
Then the documentation should be updated to remove the confusion about expected results.
Documentation should also be updated to demonstrate how to trigger refreshing of a parent route segment in NextJS 13 idiomatic way.

### Which browser are you using? (if relevant)

irrelevant

### How are you deploying your application? (if relevant)

irrelevant

**Comment by ThisIsSimmon** [comment]
Hi @dream-ellie 

They are planning to implement the ability to revalidate specific paths in the next version.
Maybe you can do what you want to do if this feature is implemented?🤔

> Next up, we plan to implement programmatic updates to specific paths or cache tags.
> ```.ts
>revalidatePath(`/posts/${slug}`);
>revalidatePath(`/posts/list`);
>revalidateTag(`post-${id}`);
>```

https://vercel.com/blog/vercel-cache-api-nextjs-cache#:~:text=Next%20up%2C%20we%20plan%20to%20implement%20programmatic%20updates%20to%20specific%20paths%20or%20cache%20tags.
